kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/EnvelopeEncryptionFilterTest.java

```diff
@@ -301,4 +301,21 @@ public void describeTo(org.hamcrest.Description description) {
             }
         });
     }
-}
+
+    private static class StubKekSelector<K> extends TopicNameBasedKekSelector<K> {
+        private final Map<String, EncryptionScheme<K>> encryptionSchemesByTopicName = new ConcurrentHashMap<>();
+
+        public void addMapping(String topicName, EncryptionScheme<K> encryptionScheme) {
+            encryptionSchemesByTopicName.put(topicName, encryptionScheme);
+        }
+
+        @NonNull
+        @Override
+        public CompletionStage<Map<String, EncryptionScheme<K>>> selectKek(@NonNull Set<String> topicNames) {
+            return CompletableFuture.completedFuture(encryptionSchemesByTopicName.entrySet()
+                    .stream()
+                    .filter(entry -> topicNames.contains(entry.getKey()))
```

Comment:
The lack of this filer was leading to my original issue however it was much harder to fix with a semantically valid `null`.